test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs

```diff
@@ -1900,5 +1901,41 @@ private static string GetTemporaryDirectory()
             Directory.CreateDirectory(tempDirectory);
             return tempDirectory;
         }
+
+        [TensorFlowFact]
+        public void TensorflowPlaceholderShapeInferenceTest()
+        {
+            //download the pb file
+            string gitPath = "https://github.com/dotnet/machinelearning-testdata/raw/ddf00f8d60ff1b4bb1f3e33639bec318e67fd32e/Microsoft.ML.TensorFlow.TestModels/cifar_model/frozen_model_variadic_input_shape.pb";
+            using (WebClient client = new WebClient())
+            {
+                client.DownloadFile(new Uri($"{gitPath}"), "cifar_model/frozen_model_variadic_input_shape.pb");
+            }
+            //frozen_model_variadic_input_shape.pb is modified by frozen_model.pb 
+            //the shape of placeholder is changed from [?, w, h, c] to [?, ?, ?, c]
+            string modelLocation = "cifar_model/frozen_model_variadic_input_shape.pb";
+
+            int imageHeight = 32;
+            int imageWidth = 32;
+            string dataFile = GetDataPath("images/images.tsv");
+            string imageFolder = Path.GetDirectoryName(dataFile);
+
+            IDataView data = _mlContext.Data.LoadFromTextFile(dataFile, new[] {
+                new TextLoader.Column("imagePath", DataKind.String, 0),
+                new TextLoader.Column("name", DataKind.String, 1)
+            });
+
+            var pipeline = _mlContext.Transforms.LoadImages("Input", imageFolder, "imagePath")
+                .Append(_mlContext.Transforms.ResizeImages("Input", imageHeight, imageWidth))
+                .Append(_mlContext.Transforms.ExtractPixels("Input", interleavePixelColors: true))
+                .Append(_mlContext.Model.LoadTensorFlowModel(modelLocation).ScoreTensorFlowModel("Output", "Input"));
+
+            var transformer = pipeline.Fit(data);
+
+            Tensorflow.TensorShape[] tfInputShape = transformer.LastTransformer.TFInputShapes;
```

Comment:
When you use LoadTensorFlowModel you need an 'using' clause to safely dispose of the unmanaged memory. 
Please see examples above.

Reply:
Thanks for reminding! I see that we are using "using clause" for nearly all LoadTensorFlowModel(). Is it inevitably memory leakage prone?